tests/codegen/abi-win64-zst.rs

```diff
@@ -0,0 +1,50 @@
+//@ compile-flags: -Z merge-functions=disabled
+
+//@ revisions: linux
+//@[linux] compile-flags: --target x86_64-unknown-linux-gnu
+//@[linux] needs-llvm-components: x86
+
+//@ revisions: windows-gnu
+//@[windows-gnu] compile-flags: --target x86_64-pc-windows-gnu
+//@[windows-gnu] needs-llvm-components: x86
+
+//@ revisions: windows-msvc
+//@[windows-msvc] compile-flags: --target x86_64-pc-windows-msvc
+//@[windows-msvc] needs-llvm-components: x86
+
+#![feature(no_core, lang_items, rustc_attrs, abi_vectorcall)]
+#![no_core]
+#![crate_type = "lib"]
+
+#[lang = "sized"]
+trait Sized {}
+
+// Make sure the argument is always passed when explicitly requesting a Windows ABI.
+
+// CHECK: define win64cc void @pass_zst_win64(ptr {{.*}})
+#[no_mangle]
+extern "win64" fn pass_zst_win64(_: ()) {}
```

Comment:
it just occurred to me, can you add some tests for returning ZSTs other than `()`? those may also need to be changed to match clang...
https://clang.godbolt.org/z/hKv74ThnE

Reply:
Oh, those are also returned by-ptr then? Wow what an odd ABI.

It seems a bit strange to special-case `()` for return types and not for argument types, though... also, it can't be just `()`. A `repr(transparent)` newtype around `()` is guaranteed to be ABI-compatible. In fact, we are saying

"Any two types with size 0 and alignment 1 are ABI-compatible."

So... I think we have a contradiction? It's not possible to satisfy all these constraints together. Windows never ceases to amaze, and this time it's not even Microsoft's fault...

Reply:
Opened https://github.com/rust-lang/unsafe-code-guidelines/issues/552 for this.
Maybe we can make this PR just about arguments? The issue around return types is pre-existing, and orthogonal to the issue around non-"C" ABIs.

Reply:
> Maybe we can make this PR just about arguments?

considering https://github.com/rust-lang/unsafe-code-guidelines/issues/552, I think just having this PR be about arguments is good enough for now, return types can be fixed later once UCG makes a decision.